lib/atlas/dataset/derived_dataset.rb

```diff
@@ -0,0 +1,27 @@
+module Atlas
+  class Dataset::DerivedDataset < Dataset
+    GRAPH     = 'graph.yml'.freeze
+
+    attribute :base_dataset,      String
+    attribute :scaling,           Hash
+
+    validate :base_dataset_exists
+
+    def graph
+      YAML.load_file(File.join(directory, area, GRAPH))
+    end
+
+    # Overwrite
+    def dataset_dir
+      @dataset_dir ||= Atlas.data_dir.join(DIRECTORY).join(Dataset::FullDataset.find(base_dataset).key.to_s)
+    end
+
+    #######
```

Comment:
FYI We're not putting `#######` around `private` in new code. Atlas hasn't been updated to reflect this, but I think new code shouldn't do it even if it is consistent.